### ISSUE TITLE
[StructuredList] Unable to navigate with keyboard

### ISSUE THREAD

**Issue by rob-richards** [carbon-design-system/carbon#2820]
## Detailed description

In v9:
Tabbing through the list does not work at all.

In v10:
When trying to navigate the `<StructuredList>` component, I am unable to press the enter or space key to select the row item.

**Comment by dakahn** [comment]
Hi 👋 Thanks for submitting and issue. I can confirm the lack of selection on keydown in v10 for enter using our example site, but could you provide the information requested in the template?

**Comment by emyarod** [comment]
just to confirm is this just for v9 and v10 react only? vanilla v10 structured list should already have keyboard navigation https://github.com/carbon-design-system/carbon/pull/1427

**Comment by dhyliu** [comment]
@emyarod According to @zgrossbart The issue was found in the product which is using carbon-components version 9.91.4 (this is a v9 version) and carbon-components-react version 6.115.5.

**Comment by emyarod** [comment]
I see, yeah it looks like this has been a missing feature from the React component for a while and includes the current version as well. we will likely need to refactor this component to support keyboard navigation

**Comment by dhyliu** [comment]
@emyarod when do you think there can be fix available for my accessibility tester to verify? Any target date for resolution? Please advise.

**Comment by emyarod** [comment]
no ETA for a bugfix at the moment but the next Carbon release would be in August

**Comment by xubinnb** [comment]
Hi @emyarod, any updates for this issue? We need a fix for v9, thanks.

**Comment by dakahn** [comment]
Escalating this after confirmation that we have a AVT2 and AVT3 violation. Cannot make a selection in or out of screen reader.

**Comment by joshblack** [comment]
Closed by https://github.com/carbon-design-system/carbon/pull/8172
